src/_P084_VEML6070.ino

```diff
@@ -0,0 +1,230 @@
+
+        if (isnan(uv_raw) || uv_raw == (-1) || uv_raw == 65535) {
+          String log = F("VEML6070: no data read!");
+          addLog(LOG_LEVEL_INFO, log);
+          UserVar[event->BaseVarIndex + 0] = NAN;
+          UserVar[event->BaseVarIndex + 1] = NAN;
+          UserVar[event->BaseVarIndex + 2] = NAN;
+		      success = false;
+        } else {
+          UserVar[event->BaseVarIndex + 0] = uv_raw;
+          UserVar[event->BaseVarIndex + 1] = uv_risk;
+          UserVar[event->BaseVarIndex + 2] = uv_power;
+          String log = F("VEML6070: UV: ");
+          log += UserVar[event->BaseVarIndex];
+          addLog(LOG_LEVEL_INFO, log);
+		      success = true;
+        }
+
+        success = true;
+        break;
+      }
+
+  }
+  return success;
+}
+
+
+//////////////
+// VEML6070 //
+//////////////
+
+//get UV raw values
+uint16_t VEML6070_ReadUv(void)
+{
+  uint16_t uv_raw = 0;
+  // read high byte
+  if (Wire.requestFrom(VEML6070_ADDR_H, 1) != 1) {
+    return -1;
+  }
+  uv_raw   = Wire.read();
+  uv_raw <<= 8;
+  // read low byte
+  if (Wire.requestFrom(VEML6070_ADDR_L, 1) != 1) {
+    return -1;
+  }
+  uv_raw  |= Wire.read();
+  // high and low done
+  return uv_raw;
+
+}
+
+
+bool VEML6070_Init(byte it)
+{
+  byte error;
+  
+  Wire.begin();
+  Wire.beginTransmission(VEML6070_ADDR_L);
+  Wire.write((it << 2) | 0x02);
+  error = Wire.endTransmission();
+  delay(500);
+
+  return (error == 0);
+}
+
+double VEML6070_UvRiskLevel(uint16_t uv_level)
+{
+  double risk = 0;
+  if (uv_level < uv_risk_map[VEML6070_UV_MAX_INDEX-1]) {
+    risk = (double)uv_level / uv_risk_map[0];
+    // generate uv-risk string
+    if ( (risk >= 0) && (risk <= 2.9) ) { snprintf_P(str_uvrisk_text, sizeof(str_uvrisk_text), D_UV_INDEX_1); }
+    else if ( (risk >= 3.0)  && (risk <= 5.9) )  { snprintf_P(str_uvrisk_text, sizeof(str_uvrisk_text), D_UV_INDEX_2); }
+    else if ( (risk >= 6.0)  && (risk <= 7.9) )  { snprintf_P(str_uvrisk_text, sizeof(str_uvrisk_text), D_UV_INDEX_3); }
+    else if ( (risk >= 8.0)  && (risk <= 10.9) ) { snprintf_P(str_uvrisk_text, sizeof(str_uvrisk_text), D_UV_INDEX_4); }
+    else if ( (risk >= 11.0) && (risk <= 12.9) ) { snprintf_P(str_uvrisk_text, sizeof(str_uvrisk_text), D_UV_INDEX_5); }
+    else if ( (risk >= 13.0) && (risk <= 25.0) ) { snprintf_P(str_uvrisk_text, sizeof(str_uvrisk_text), D_UV_INDEX_6); }
+    else { snprintf_P(str_uvrisk_text, sizeof(str_uvrisk_text), D_UV_INDEX_7); }
+    return risk;
+  } else {
+    // out of range and much to high - it must be outerspace or sensor damaged
+    snprintf_P(str_uvrisk_text, sizeof(str_uvrisk_text), D_UV_INDEX_7);
+    String log = F("VEML6070 out of range: ");
+    log += risk;
+    addLog(LOG_LEVEL_DEBUG, log);
+    return ( risk = 99 );
+  }
+}
+
+double VEML6070_UvPower(double uvrisk)
+{
+  // based on calculations for effective irradiation from Karel Vanicek
+  double power = 0;
+  return ( power = VEML6070_POWER_COEFFCIENT * uvrisk );
```

Comment:
Why create a `double power` here and not return that value.
Simply `return VEML6070_POWER_COEFFCIENT * uvrisk;`

Reply:
I agree, changed to suggested